api.c

```diff
@@ -348,6 +349,32 @@ int tcmu_emulate_evpd_inquiry(
 		return SAM_STAT_GOOD;
 	}
 	break;
+	case 0x80: /* Unit Serial Number */
+	{
+		char data[512];
+		char *wwn;
+		uint32_t len;
+
+		memset(data, 0, sizeof(data));
+
+		data[1] = 0x80;
+
+		wwn = tcmu_get_wwn(dev);
+		if (!wwn) {
+			return tcmu_set_sense_data(sense, HARDWARE_ERROR,
+						   ASC_INTERNAL_TARGET_FAILURE,
+						   NULL);
+		}
+
+		len = sprintf(&data[4], "%s", wwn);
```

Comment:
How about snprintf here?  Just ensures that in case the wwn ever mysteriously returns > 508 bytes, you won't corrupt the stack.